-fcompiler-rt broken with object file builds

### Zig Version

0.11.0-dev.980+601ab9a25

### Steps to Reproduce and Observed Behavior


Build command:
1. `zig build-obj main.zig --cache-dir /home/jarred/Build/obj/zig-cache --global-cache-dir /home/jarred/.cache/zig --name Build -fcompiler-rt --enable-cache`
2. `clang build/Build.o`

Source file:
```zig
const std = @import("std");

pub export fn hey() void {}

pub fn main() !void {
    // Prints to stderr (it's a shortcut based on `std.io.getStdErr()`)
    std.debug.print("All your {s} are belong to us.\n", .{"codebase"});

    // stdout is for the actual output of your application, for example if you
    // are implementing gzip, then only the compressed bytes should be sent to
    // stdout, not any debugging messages.
    const stdout_file = std.io.getStdOut().writer();
    var bw = std.io.bufferedWriter(stdout_file);
    const stdout = bw.writer();

    try stdout.print("Run `zig build test` to run the tests.\n", .{});

    try bw.flush(); // don't forget to flush!
}

test "simple test" {
    var list = std.ArrayList(i32).init(std.testing.allocator);
    defer list.deinit(); // try commenting this out and see if zig detects the memory leak!
    try list.append(42);
    try std.testing.expectEqual(@as(i32, 42), list.pop());
}
```

Output:
```zig
❯ clang build/Build.o
/usr/bin/ld: build/Build.o: in function `_start':
/home/jarred/zig/0.11.0-dev.980+601ab9a25/files/lib/std/start.zig:305: multiple definition of `_start'; /lib/x86_64-linux-gnu/crt1.o:(.text+0x0): first defined here
/usr/bin/ld: /lib/x86_64-linux-gnu/crt1.o: in function `_start':
(.text+0x1b): undefined reference to `main'
/usr/bin/ld: build/Build.o: in function `debug.panicExtra__anon_1096':
/home/jarred/zig/0.11.0-dev.980+601ab9a25/files/lib/std/debug.zig:301: undefined reference to `__zig_probe_stack'
/usr/bin/ld: build/Build.o: in function `main.main':
/home/jarred/Build/obj/src/main.zig:5: undefined reference to `__zig_probe_stack'
/usr/bin/ld: build/Build.o: in function `debug.panicExtra__anon_3868':
/home/jarred/zig/0.11.0-dev.980+601ab9a25/files/lib/std/debug.zig:301: undefined reference to `__zig_probe_stack'
/usr/bin/ld: build/Build.o: in function `debug.readElfDebugInfo':
/home/jarred/zig/0.11.0-dev.980+601ab9a25/files/lib/std/debug.zig:890: undefined reference to `__zig_probe_stack'
/usr/bin/ld: build/Build.o: in function `fs.Dir.openFile':
/home/jarred/zig/0.11.0-dev.980+601ab9a25/files/lib/std/fs.zig:1099: undefined reference to `__zig_probe_stack'
/usr/bin/ld: build/Build.o:/home/jarred/zig/0.11.0-dev.980+601ab9a25/files/lib/std/debug.zig:1115: more undefined references to `__zig_probe_stack' follow
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

### Expected Behavior

No build errors, successful compilation

I had the same problem when trying to build a static executable with a C compiler from the https://ziglang.org/documentation/master/#Exporting-a-C-Library example.

Here are the steps:
```sh
zig build-lib -femit-h mathtest.zig

gcc -o test -static -I $(zig env | jq -r .lib_dir) test.c -L . -lmathtest
./test
```

```
/usr/bin/ld: ./libmathtest.a(./libmathtest.a.o): in function `debug.readElfDebugInfo':
/home/manlio/.local/share/sdk/zig/master/lib/std/debug.zig:939: undefined reference to `__zig_probe_stack'
/usr/bin/ld: ./libmathtest.a(./libmathtest.a.o): in function `fs.Dir.openFile':
/home/manlio/.local/share/sdk/zig/master/lib/std/fs.zig:1099: undefined reference to `__zig_probe_stack'
/usr/bin/ld: ./libmathtest.a(./libmathtest.a.o): in function `debug.printLineFromFileAnyOs__anon_5459':
/home/manlio/.local/share/sdk/zig/master/lib/std/debug.zig:1164: undefined reference to `__zig_probe_stack'
/usr/bin/ld: ./libmathtest.a(./libmathtest.a.o): in function `dwarf.DwarfInfo.getLineNumberInfo':
/home/manlio/.local/share/sdk/zig/master/lib/std/dwarf.zig:1090: undefined reference to `__zig_probe_stack'
/usr/bin/ld: ./libmathtest.a(./libmathtest.a.o): in function `debug.panicExtra__anon_5748':
/home/manlio/.local/share/sdk/zig/master/lib/std/debug.zig:297: undefined reference to `__zig_probe_stack'
/usr/bin/ld: ./libmathtest.a(./libmathtest.a.o):/home/manlio/.local/share/sdk/zig/master/lib/std/debug.zig:297: more undefined references to `__zig_probe_stack' follow
collect2: error: ld returned 1 exit status
```

Using dynamic linking works:
```sh
zig build-lib -femit-h mathtest.zig -dynamic

gcc -o test -I $(zig env | jq -r .lib_dir) test.c -L . -lmathtest
LD_LIBRARY_PATH="." ./test
```

--- a/src/Compilation.zig
+++ b/src/Compilation.zig
@@ -1635,9 +1635,24 @@ pub fn create(gpa: Allocator, options: InitOptions) !*Compilation {
             } else main_pkg;
             errdefer if (options.is_test) root_pkg.destroy(gpa);
 
+            const compiler_rt_pkg = if (include_compiler_rt and options.output_mode == .Obj) compiler_rt_pkg: {
+                break :compiler_rt_pkg try Package.createWithDir(
+                    gpa,
+                    "compiler_rt",
+                    options.zig_lib_directory,
+                    null,
+                    "compiler_rt.zig",
+                );
+            } else null;
+            errdefer if (compiler_rt_pkg) |p| p.destroy(gpa);
+
             try main_pkg.addAndAdopt(gpa, builtin_pkg);
             try main_pkg.add(gpa, root_pkg);
             try main_pkg.addAndAdopt(gpa, std_pkg);
+
+            if (compiler_rt_pkg) |p| {
+                try main_pkg.addAndAdopt(gpa, p);
+            }
 
             const main_pkg_is_std = m: {
                 const std_path = try std.fs.path.resolve(arena, &[_][]const u8{
@@ -2355,6 +2370,10 @@ pub fn update(comp: *Compilation) !void {
             _ = try module.importPkg(module.main_pkg);
         }
 
+        if (module.main_pkg.table.get("compiler_rt")) |compiler_rt_pkg| {
+            _ = try module.importPkg(compiler_rt_pkg);
+        }
+
         // Put a work item in for every known source file to detect if
         // it changed, and, if so, re-compute ZIR and then queue the job
         // to update it.
@@ -2378,6 +2397,10 @@ pub fn update(comp: *Compilation) !void {
         try comp.work_queue.writeItem(.{ .analyze_pkg = std_pkg });
         if (comp.bin_file.options.is_test) {
             try comp.work_queue.writeItem(.{ .analyze_pkg = module.main_pkg });
+        }
+
+        if (module.main_pkg.table.get("compiler_rt")) |compiler_rt_pkg| {
+            try comp.work_queue.writeItem(.{ .analyze_pkg = compiler_rt_pkg });
         }
     }
 

